python_modules/dagster/dagster/_core/execution/plan/execute_step.py

```diff
@@ -873,10 +868,26 @@ def _gen_fn():
         )
 
 
-def _log_asset_materialization_events_for_asset(
-    step_context, output_context, output, output_def, manager_metadata
-):
-    asset_key, partitions = _materializing_asset_key_and_partitions_for_output(output_context)
+def _log_asset_events_for_asset(
+    step_context: StepExecutionContext,
+    output_context: OutputContext,
+    output: Union[Output, DynamicOutput],
+    output_def: OutputDefinition,
+    manager_metadata: Mapping[str, MetadataValue],
+) -> Iterator[DagsterEvent]:
+    # This is a temporary workaround to prevent duplicate observation events from external
+    # observable assets that were auto-converted from source assets. These assets yield
+    # observation events through the context in their body, and will continue to do so until we
+    # can convert them to using ObserveResult, which requires a solution to partition-scoped
+    # metadata and data version on output. We identify these auto-converted assets by looking
+    # for OBSERVATION-type asset that have this special metadata key (added in
+    # `wrap_source_asset_observe_fn_in_op_compute_fn`), which should only occur for these
+    # auto-converted source assets. This can be removed when source asset observation functions
+    # are converted to use ObserveResult.
+    if SYSTEM_METADATA_KEY_SOURCE_ASSET_OBSERVATION in output.metadata:
+        return
```

Comment:
This block needed to be moved up from below to avoid triggering an error on `_get_asset_key_and_partitions_for_output`.